assets/src/components/map/markers/stopMarker.tsx

```diff
@@ -0,0 +1,305 @@
+import React, { useMemo, useState } from "react"
+
+import { LeafletEventHandlerFnMap, PointTuple } from "leaflet"
+import "leaflet.fullscreen"
+import { MarkerProps } from "react-leaflet"
+
+import { joinClasses } from "../../../helpers/dom"
+import { DirectionId, Stop } from "../../../schedule"
+import { MobileFriendlyTooltip } from "../../mapMarkers"
+import StopCard from "../../stopCard"
+import { ReactMarker } from "../utilities/reactMarker"
+
+/**
+ * Specific variants that the `StopIcon` can render
+ */
+export enum StopType {
+  Small,
+  Medium,
+  Large,
+}
+
+/**
+ * Produces the StopType assigned to the provided map Zoom-Level
+ * @param zoomLevel The Map Zoom Level to use for figuring out the StopType
+ */
+export const StopTypeFromZoomLevel = (zoomLevel: number) => {
+  if (zoomLevel <= 14) {
+    return StopType.Small
+  } else if (15 <= zoomLevel && zoomLevel <= 16) {
+    return StopType.Medium
+  } else if (17 <= zoomLevel) {
+    return StopType.Large
+  } else {
+    throw new Error(
+      "internal error: entered unreachable code: invalid zoom level"
+    )
+  }
+}
+
+export const StopIconSizeFromZoomLevel = (zoomLevel: number) =>
+  StopIconSizeFromStopType(StopTypeFromZoomLevel(zoomLevel))
+
+/**
+ * Represents the box size of the icon at each `StopType`.
+ *
+ * For a `<circle>`, this means the size needs to include the `radius` AND
+ * `stroke-width`, or specify `overflow-visible`. This will directly impact
+ * the tap target on the leaflet map.
+ */
+export const StopIconSizeFromStopType = (stopType: StopType): PointTuple => {
```

Comment:
**nitpick:** since this isn't a component, shouldn't it be `StopIconSizeFromStopType`?
```suggestion
export const stopIconSizeFromStopType = (stopType: StopType): PointTuple => {
```

Reply:
👍🏻 19d50226388f15afff0111d6126c9fa07c194102